### ISSUE TITLE
Adding Aries-cloudagent to the Private Identity Sever back-end

### ISSUE THREAD

**Issue by H4LL** [OpenMined/aries-did.js#2]
We can give the Private Identity Server (PIS) Aries Verifiable Credentials (VCs) and Self-sovereign Identity (SSI) by allowing the PIS to talk to an Aries back-end. 

This is not dissimilar to what is being done in the current version of the project. We extend the Aries Agent back-end with a GUI web front-end. If we can apply the communication methods between the current front-end and the PIS server, this should be achievable. We need someone to investigate how we can pull this off. Specifically, by examining how this communication works and how we can appropriate it. @wip-abramson

**Comment by wip-abramson** [comment]
Yep so actually without having much knowledge of the PIS architecture I would phrase this as:

We need to turn the PIS backend into an aries **controller**. The start of this recording covers that well - https://zoom.us/recording/play/Pr-gdxAUVciy7MtPE9tkNAuSLT_Pl_NEYMeW2XxQRitjXtQajl3X5y7L_A1CCRee?autoplay=true&startTime=1563894122000

Additionally, we need to run an Aries agent - this can be done through the command line but should be invoked through code. 

The controller holds all the business logic and communicate with the agent over http (although other protocols are possible) whenever it wants some SSI specific functionality. All the capabilities of an aries agent are described as RFCs in this repo https://github.com/hyperledger/aries-rfcs/tree/master/features.

NOTE: We do not need to develop an agent. We need to develop a controller that tells a generic agent to perform and respond to agent actions with logic specific to this use case.

While there are a lot of open source agents, wading through the code can be challenging. I would recommend exploring a more polished developer friendly set of toolings https://developer.streetcred.id/. Things will go a lot smoother. Plus anything developed using this is interoperable with agents implemented with the open source code bases like aca-py.
